docs/source/deep-learning/pytorch/quickstart/pytorch_quickstart.ipynb

```diff
@@ -0,0 +1,1361 @@
+   "outputs": [
+    {
+     "output_type": "execute_result",
+     "data": {
+      "text/plain": [
+       "<Experiment: artifact_location='dbfs:/databricks/mlflow-tracking/1078557169589361', creation_time=1703121702068, experiment_id='1078557169589361', last_update_time=1703194525608, lifecycle_stage='active', name='/mlflow-pytorch-quickstart', tags={'mlflow.experiment.sourceName': '/mlflow-pytorch-quickstart',\n",
+       " 'mlflow.experimentType': 'MLFLOW_EXPERIMENT',\n",
+       " 'mlflow.ownerEmail': 'qianchen94era@gmail.com',\n",
+       " 'mlflow.ownerId': '3209978630771139'}>"
+      ]
+     },
+     "metadata": {},
+     "execution_count": 8
+    }
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "## Implement the Training Loop\n",
+    "\n",
+    "Now let's define the training loop, which basically is going through the dataset, and apply a forward and backward pass on each data batch."
+   ],
+   "metadata": {
+    "id": "TkyEv1qa6Uxe"
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "Get the device info, as PyTorch requires manual device management."
+   ],
+   "metadata": {
+    "id": "YXAdFfTa6tkH"
+   }
+  },
+  {
+   "cell_type": "code",
+   "source": [
+    "# Get cpu or gpu for training.\n",
+    "device = \"cuda\" if torch.cuda.is_available() else \"cpu\""
+   ],
+   "metadata": {
+    "id": "bVLaZw396puE"
+   },
+   "execution_count": 9,
+   "outputs": []
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "Define the training function."
+   ],
+   "metadata": {
+    "id": "YULTT1EI6_1L"
+   }
+  },
+  {
+   "cell_type": "code",
+   "source": [
+    "def train(dataloader, model, loss_fn, metrics_fn, optimizer, epoch):\n",
+    "    model.train()\n",
+    "    for batch, (X, y) in enumerate(dataloader):\n",
+    "        X, y = X.to(device), y.to(device)\n",
+    "\n",
+    "        pred = model(X)\n",
+    "        loss = loss_fn(pred, y)\n",
+    "        accuracy = metrics_fn(pred, y)\n",
+    "\n",
+    "        # Backpropagation.\n",
+    "        loss.backward()\n",
+    "        optimizer.step()\n",
+    "        optimizer.zero_grad()\n",
+    "\n",
+    "        if batch % 100 == 0:\n",
+    "            loss, current = loss.item(), batch\n",
+    "            step = batch // 100 * (epoch + 1)\n",
+    "            mlflow.log_metric(\"loss\", f\"{loss:2f}\", step=step)\n",
+    "            mlflow.log_metric(\"accuracy\", f\"{accuracy:2f}\", step=step)\n",
+    "            print(f\"loss: {loss:2f} accuracy: {accuracy:2f} [{current} / {len(dataloader)}]\")"
+   ],
+   "metadata": {
+    "id": "LDiFPTGR5ybY"
+   },
+   "execution_count": 10,
+   "outputs": []
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "Define the evaluation function, which will be run at the end of each epoch."
+   ],
+   "metadata": {
+    "id": "Z6LeQOrS7C98"
+   }
+  },
+  {
+   "cell_type": "code",
+   "source": [
+    "def evaluate(dataloader, model, loss_fn, metrics_fn, epoch):\n",
```

Comment:
It might be helpful to add pydoc strings to this function so that readers can learn a little bit about how these functions work with torch and MLflow

Reply:
good call, added